### DISCUSSION TITLE
Possible to use second SPI on ESP32-C3 ?

### DISCUSSION THREAD

**Discussion by svofski** [micropython/micropython#15309] [ESP32] [answered]
Hi!

I was going to hook up a second SPI device and I don't want to share the bus. ESP32-C3 has 3 SPI units. Micropython ESP32 docs say that SPI(1) and SPI(2) can be used. However when I try to create SPI(2), I get `ValueError: SPI(2) doesn't exist`. 

I'm using `MicroPython v1.24.0-preview.34.g5903ee561 on 2024-06-08; ESP32C3 module with ESP32C3` on WaveShare ESP32-C3-Zero.

Is this expected?

**Comment by ricksorensen** [comment, marked as answer]
On the ESP32C3 SPI 0 & 1 are used for memory (flash) devices, SPI2 is a general purpose Flash.
If you want to use multiple SPI busses- try SoftSPI.  The micropython doc for the ESP32 is for the the ESP32, and coud be amended for the other varieties.

From the Espressif C3 Datasheet:
```
3.4.2 Serial Peripheral Interface (SPI)
ESP32-C3 has the following SPI interfaces:
• SPI0 used by ESP32-C3’s GDMA controller and cache to access in-package or off-package flash
• SPI1 used by the CPU to access in-package or off-package flash
• SPI2 is a general purpose SPI controller with access to a DMA channel allocated by the GDMA controller
```

**Reply by svofski** [reply]
Thank you, that makes it clear. I agree that the documentation could mention C3 differences if it doesn't yet do that.

**Comment by QGB** [comment]
esp32c3 SPI(2) doesn't exist
